sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIO.java

```diff
@@ -1447,27 +1453,25 @@ public void processElement(@Element PubsubMessage message, @Timestamp Instant ti
         }
 
         // NOTE: The record id is always null.
-        currentTopicOutput.add(
+        currentTopicOutput.messages.add(
             OutgoingMessage.of(
                 msgBuilder.build(), timestamp.getMillis(), null, message.getTopic()));
-        currentOutputBytes.put(pubsubTopic, currentTopicOutputBytes + messageSize);
+        currentTopicOutput.bytes += messageSize;
       }
 
       @FinishBundle
       public void finishBundle() throws IOException {
         publish();
         output = null;
-        currentOutputBytes = null;
         pubsubClient.close();
         pubsubClient = null;
       }
 
       private void publish() throws IOException {
```

Comment:
inline this in finishBundle? otherwise the topic specific publish should be called